src/js/tabs/ports.js

```diff
@@ -351,50 +350,48 @@ TABS.ports.initialize = function (callback, scrollPosition) {
 
    function on_save_handler() {
         analytics.sendChangeEvents(analytics.EVENT_CATEGORIES.FLIGHT_CONTROLLER, self.analyticsChanges);
-       self.analyticsChanges = {};
+        self.analyticsChanges = {};
 
         // update configuration based on current ui state
         FC.SERIAL_CONFIG.ports = [];
 
-        var ports_e = $('.tab-ports .portConfiguration').each(function (portConfiguration_e) {
-
-            var portConfiguration_e = this;
-
-            var oldSerialPort = $(this).data('serialPort');
+        $('.tab-ports .portConfiguration').each(function (portConfiguration) {
+            const portConfig = portConfiguration;
+            const oldSerialPort = $(this).data('serialPort');
```

Comment:
`const oldSerialPort = portConfiguration.data('serialPort')` seems to be easier to understand, since we are not using `this` in the rest of this closure.

Reply:
Thanks. Will revisit.

Reply:
Fixed.
`portConfiguration` is a number.
Started from scratch, rebased & tested.

Reply:
`PortConfig`  was previous a number and it was fixed using `this`

💡The first function argument is the key function argument. We needed a second function argument which contains the configuration elements we want, so `this` is not needed anymore.